File: backend/futdelas/src/main/java/org/ifb/tcc/futdelas/entities/UsuarioNotificacao.java
package org.ifb.tcc.futdelas.entities;

import java.time.LocalDateTime;
import java.util.UUID;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
@Entity
public class UsuarioNotificacao {
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;
    private String deviceToken;

    private Integer timeId;

    private boolean notificacoesAtivas;

    private LocalDateTime ultimaNotificacaoEnviada;
}

File: frontend/futedelas-app/src/api/interfaces/UsuarioNotificacao.ts
export class UsuarioNotificacao {
  id: number | undefined;
  deviceToken: string;
  timeId: number;
  ativo: boolean;

  constructor(deviceToken: string, timeId: number, ativo: boolean) {
    this.deviceToken = deviceToken;
    this.timeId = timeId;
    this.ativo = ativo;
  }
}